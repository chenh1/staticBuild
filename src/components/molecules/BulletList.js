import React from 'react';
import ListItem from '../atoms/ListItem';

const BulletList = ({styleClass = "list-one", listItems}) => {
    return (
        <ul className={styleClass}>
            {listItems.map((listItem, index) =>
                <ListItem 
                    key={"li" + index} 
                    text={listItem.text || listItem} 
                    path={listItem.path}
                    linkClass={listItem.linkClass}
                    linkActiveClass={listItem.linkActiveClass}
                /> 
            )}
        </ul>
    );
};

export default BulletList;