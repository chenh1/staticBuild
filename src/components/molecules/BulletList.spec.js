import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import BulletList from './BulletList';

test('renders BulletList with text', () => {
    const tree = renderer.create(<BulletList
        listItems={[
            'list item 1',
            'list item 2',
            'list item 3'
        ]}
        customClass="mock-class" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

test('renders BulletList with default list-one class', () => {
    const tree = renderer.create(<BulletList
        listItems={[
            'list item 1',
            'list item 2',
            'list item 3'
        ]} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});