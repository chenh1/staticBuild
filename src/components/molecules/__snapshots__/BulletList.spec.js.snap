// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BulletList with default list-one class 1`] = `
<ul
  className="list-one"
>
  <li>
    <span>
      list item 1
    </span>
  </li>
  <li>
    <span>
      list item 2
    </span>
  </li>
  <li>
    <span>
      list item 3
    </span>
  </li>
</ul>
`;

exports[`renders BulletList with text 1`] = `
<ul
  className="list-one"
>
  <li>
    <span>
      list item 1
    </span>
  </li>
  <li>
    <span>
      list item 2
    </span>
  </li>
  <li>
    <span>
      list item 3
    </span>
  </li>
</ul>
`;
